spec/unit/shoulda/matchers/active_record/have_rich_text_matcher_spec.rb

```diff
@@ -0,0 +1,137 @@
+require 'unit_spec_helper'
+
+describe Shoulda::Matchers::ActiveRecord::HaveRichText, type: :model do
+  def self.rich_text_is_defined?
+    defined?(ActionText::RichText)
+  end
+
+  if rich_text_is_defined?
+    it 'returns the message including the name of the provided attribute' do
+      matcher = have_rich_text(:content)
+      expect(matcher.description).
+        to eq('have :content as a ActionText::RichText')
+    end
+
+    context 'a has_rich_text association' do
+      it 'matches when the subject configures has_rich_text for a custom ' \
+        'attribute' do
+        create_table_posts
+    
+        valid_record = new_post(has_rich_text: true)
+    
+        expect(valid_record).to have_rich_text(:content)
+      end
+    
+      it 'does not matches when lacking has_rich_text ' \
+          'attribute' do
+        create_table_posts
+    
+        invalid_record = new_post(has_rich_text: false)
+        expected_message = 'Expected Post to have :invalid_attribute as a ' \
+          'ActionText::RichText but :invalid_attribute does not exist'
+    
+        aggregate_failures do
```

Comment:
Instead of using `aggregate_failures` and having to repeat yourself twice, there's a new-ish matcher you can use called `match_against` which checks that a matcher matches when used in the negative and does not match when used in the positive. You can see that in use here: https://github.com/thoughtbot/shoulda-matchers/blob/0011a3bd9a6b07f8234954d0418a74bdeb393db7/spec/unit/shoulda/matchers/active_model/validate_presence_of_matcher_spec.rb#L529-L536. So in this case you would say:

``` ruby
expect { have_rich_text(:invalid_attribute) }.
  not_to match_against(invalid_record).
  and_fail_with(expected_message)
```

By the way, if you want to thoroughly test both versions of the failure message (`failure_message` and `failure_message_when_negated`), you can also do the same thing with the positive cases, so you could say:

``` ruby
expect { have_rich_text(:content) }.
  to match_against(valid_record).
  or_fail_with(expected_message)
```

and this will test that the matcher matches when used in the positive and does not match when used in the negative.

Reply:
Much better! Thanks!